### PR TITLE
Add java interfaces and implementations for stateful checkpoint node

### PR DESCRIPTION
**Pull request by maosongfu**

### CODE REVIEW

File: heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java [hunk 1]
```diff
@@ -182,6 +185,18 @@ public Lock getLock(String topologyName, LockName lockName) {
   }
 
   @Override
+  public ListenableFuture<CheckpointManager.StatefulMostRecentCheckpoint> getStatefulCheckpoint(
```
Comment (srkukarni) [inline heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java]: where is the setStatefulCheckpoint override?

File: heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java [hunk 2]
```diff
@@ -213,6 +214,12 @@ protected boolean initTree() {
   }
 
   @Override
+  public ListenableFuture<Boolean> setStatefulCheckpoint(
```
Comment (srkukarni) [inline heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java]: I don't understand this. Why are there multiple files?

File: heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java (hunk 1 above)
Comment (maosongfu) [inline heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java]: So LocalFileSystemStateManager extends FileSystemStateManager; FileSystemStateManager extends IStateManager.

This is refactored by Bill earlier. So the getXXX, deleteXXX will be overridden here while the setXXX will be overridden in the actual one. So here I followed his convention.

File: heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java (hunk 2 above)
Comment (maosongfu) [inline heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java]: So LocalFileSystemStateManager extends FileSystemStateManager; FileSystemStateManager extends IStateManager.

This is refactored by Bill earlier. So the getXXX, deleteXXX will be overridden here while the setXXX will be overridden in the actual one. So here I followed his convention.

**Pull request merged**